plugin/src/main/kotlin/org/gradle/github/dependencygraph/internal/DependencyGraphRenderer.kt

```diff
@@ -0,0 +1,14 @@
+package org.gradle.github.dependencygraph.internal
+
+import org.gradle.github.dependencygraph.internal.model.BuildLayout
+import org.gradle.github.dependencygraph.internal.model.ResolvedConfiguration
+import org.gradle.github.dependencygraph.internal.util.PluginParameters
+import java.io.File
+
+interface DependencyGraphRenderer {
+    fun outputDependencyGraph(pluginParameters: PluginParameters,
+                              buildLayout: BuildLayout,
+                              resolvedConfigurations: MutableList<ResolvedConfiguration>,
+                              outputDirectory: File
```

Comment:
Should the `outputDependencyGraph` method have the responsibility of writing the file out to disk? Wouldn't we likely want to control how that is done via a consistent API?

Reply:
Maybe. Although it's nice to allow a renderer to output multiple files. For example, you could imagine an HTML report renderer that writes a bunch of files as it's output.